OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/string/StringAnalysisState.scala

```diff
@@ -0,0 +1,215 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package tac
+package fpcf
+package analyses
+package string
+
+import scala.collection.mutable
+
+import org.opalj.br.DeclaredMethod
+import org.opalj.br.DefinedMethod
+import org.opalj.br.Method
+import org.opalj.br.fpcf.properties.Context
+import org.opalj.br.fpcf.properties.cg.Callers
+import org.opalj.br.fpcf.properties.string.StringConstancyProperty
+import org.opalj.br.fpcf.properties.string.StringTreeNode
+import org.opalj.br.fpcf.properties.string.StringTreeOr
+import org.opalj.fpcf.Entity
+import org.opalj.fpcf.EOptionP
+import org.opalj.fpcf.EPS
+import org.opalj.fpcf.Property
+import org.opalj.fpcf.PropertyStore
+import org.opalj.tac.fpcf.properties.TACAI
+import org.opalj.tac.fpcf.properties.string.MethodStringFlow
+
+private[string] case class ContextFreeStringAnalysisState(
+    entity:                 VariableDefinition,
+    var stringFlowDependee: EOptionP[Method, MethodStringFlow],
+    var hitMaximumDepth:    Boolean = false
+) {
+
+    def hasDependees: Boolean = stringFlowDependee.isRefinable
+
+    def dependees: Set[EOptionP[Entity, Property]] = Set(stringFlowDependee)
+}
+
+private[string] class ContextStringAnalysisState private (
+    _entity:                       VariableContext,
+    var _stringDependee:           EOptionP[VariableDefinition, StringConstancyProperty],
+    private var _parameterIndices: Set[Int]
+) {
+
+    def entity: VariableContext = _entity
```

Comment:
Why is this a def backed by a constructor parameter rather than just a val in the constructor?